src/julia-syntax.scm

```diff
@@ -1425,14 +1425,17 @@
             (reverse a))))))
 
 ;; lower function call containing keyword arguments
-(define (lower-kw-call f kw pa)
+(define (lower-kw-call fexpr kw pa)
   (let ((container (make-ssavalue)))
     (let loop ((kw kw)
                (initial-kw '()) ;; keyword args before any splats
                (stmts '())
                (has-kw #f))     ;; whether there are definitely >0 kwargs
       (if (null? kw)
-          (if (null? stmts)
+        (let ((f (if (sym-ref? fexpr) fexpr (make-ssavalue))))
+          `(block
+            ,(if (eq? f fexpr) () `(= ,f, fexpr))
```

Comment:
A slightly better way to write this is `,@(if (eq? f fexpr) '() `((= ,f, fexpr)))`, which will insert nothing at all when no assignment is needed.